src/main/java/com/cst438/controller/AssignmentController.java

```diff
@@ -101,18 +143,30 @@ public void updateGrades(@RequestBody List<GradeDTO> dlist) {
 
     // student lists their assignments/grades for an enrollment ordered by due date
     // student must be enrolled in the section
+    // return a list of assignments and (if they exist) the assignment grade
+    //  for all sections that the student is enrolled for the given year and semester
+    //  hint: use the assignment repository method findByStudentIdAndYearAndSemesterOrderByDueDate
     @GetMapping("/assignments")
     public List<AssignmentStudentDTO> getStudentAssignments(
             @RequestParam("studentId") int studentId,
             @RequestParam("year") int year,
             @RequestParam("semester") String semester) {
```

Comment:
Might need to throw on invalid year/semester.

Reply:
if (year < 2000 || year > 2100) {
    throw new ResponseStatusException(
        HttpStatus.BAD_REQUEST, "Year must be between 2000 and 2100"
    );
} else if (semester == null) {
    throw new ResponseStatusException(
        HttpStatus.BAD_REQUEST, "Semester cannot be null"
    );
}